core-bundle/src/Resources/contao/elements/ContentDownloads.php

```diff
@@ -99,14 +99,15 @@ protected function compile()
 		$auxDate = array();
 
 		$objFiles = $this->objFiles;
-		$request = System::getContainer()->get('request_stack')->getCurrentRequest();
+		$container = System::getContainer();
+		$request = $container->get('request_stack')->getCurrentRequest();
 		$allowedDownload = StringUtil::trimsplit(',', strtolower(Config::get('allowedDownload')));
 
 		// Get all files
 		while ($objFiles->next())
 		{
 			// Continue if the files has been processed or does not exist
-			if (isset($files[$objFiles->path]) || !file_exists(System::getContainer()->getParameter('kernel.project_dir') . '/' . $objFiles->path))
+			if (isset($files[$objFiles->path]) || !file_exists($container->getParameter('kernel.project_dir') . '/' . $objFiles->path))
```

Comment:
The `getParameter()` call is still inside a `while` loop. Wouldn‘t it make sense to define a variable here as well?

Reply:
Done. I also extracted the back end request check from the loop.